### PR TITLE
Fixed RadioBox Enable test case

### PR DESCRIPTION
**Pull request by liamtreacy**
I have changed the implementation of the `Enable(unsigned int, bool enable)` method to take account of the enabled state of the QGroupBox (if disabled, any items it holds cannot be enabled). I have also implemented the `Enable(bool enable)` method to properly match the expected wx behaviour.

### CODE REVIEW

File: src/qt/radiobox.cpp [hunk 1]
```diff
@@ -168,10 +168,42 @@ static QAbstractButton *GetButtonAt( const QButtonGroup *group, unsigned int n )
 
 bool wxRadioBox::Enable(unsigned int n, bool enable)
 {
-    QAbstractButton *qtButton = GetButtonAt( m_qtButtonGroup, n );
-    CHECK_BUTTON( qtButton, false );
+    if( enable && !m_qtGroupBox->isEnabled() )
```
Comment (catalinr) [inline src/qt/radiobox.cpp]:
```suggestion
    if ( enable && !m_qtGroupBox->isEnabled() )
```
There are some coding style guidelines that wxW uses, some of which may be unusual, but it's too late to adopt new ones so the best scenario is to be consistent and apply them in new code as well.

File: src/qt/radiobox.cpp [hunk 2]
```diff
@@ -168,10 +168,42 @@ static QAbstractButton *GetButtonAt( const QButtonGroup *group, unsigned int n )
...
+    {
+        m_qtGroupBox->setEnabled( true );
+
+        for( unsigned int i = 0; i < GetCount(); ++i )
```
Comment (catalinr) [inline src/qt/radiobox.cpp]:
```suggestion
        for ( unsigned int i = 0; i < GetCount(); ++i )
```

File: src/qt/radiobox.cpp [hunk 3]
```diff
@@ -168,10 +168,42 @@ static QAbstractButton *GetButtonAt( const QButtonGroup *group, unsigned int n )
...
+        {
+            QAbstractButton *qtButton = GetButtonAt( m_qtButtonGroup, i );
+            CHECK_BUTTON( qtButton, false );
+
+            i == n ? qtButton->setEnabled( true ) : qtButton->setEnabled( false );
```
Comment (catalinr) [inline src/qt/radiobox.cpp]:
```suggestion
            qtButton->setEnabled( i == n );
```

File: src/qt/radiobox.cpp [hunk 4]
```diff
@@ -168,10 +168,42 @@ static QAbstractButton *GetButtonAt( const QButtonGroup *group, unsigned int n )
...
+        CHECK_BUTTON( qtButton, false );
+        qtButton->setEnabled(enable);
+    }
+
+    return true;
+}
+
+bool wxRadioBox::Enable( bool enable )
+{
+    if( m_qtGroupBox->isEnabled() == enable )
```
Comment (catalinr) [inline src/qt/radiobox.cpp]:
```suggestion
    if ( m_qtGroupBox->isEnabled() == enable )
```

File: src/qt/radiobox.cpp [hunk 5]
```diff
@@ -168,10 +168,42 @@ static QAbstractButton *GetButtonAt( const QButtonGroup *group, unsigned int n )
...
+    {
+        for( unsigned int i = 0; i < GetCount(); ++i )
```
Comment (catalinr) [inline src/qt/radiobox.cpp]:
```suggestion
        for ( unsigned int i = 0; i < GetCount(); ++i )
```

**Review by catalinr — COMMENTED** [review]
coding style only

File: src/qt/radiobox.cpp (hunk 1 above)
Comment (liamtreacy) [reply src/qt/radiobox.cpp]: Thanks for the feedback, will keep that in mind for future commits.

**Comment by vadz** [comment]
Looks good, thanks, will commit soon with just the changes to the commit message wording and also splitting the `s/unsigned/unsigned int/` change in its own commit (please don't change unrelated things in the same commit).

**Pull request closed without merging**